beacon-chain/types/active_state.go

```diff
@@ -147,11 +147,12 @@ func (a *ActiveState) GetBlockVoteCache() map[[32]byte]*utils.VoteCache {
 	return a.blockVoteCache
 }
 
-// appendNewAttestations appends new attestations from block in to active state.
-// this is called during block processing.
-func (a *ActiveState) appendNewAttestations(add []*pb.AggregatedAttestation) []*pb.AggregatedAttestation {
-	existing := a.data.PendingAttestations
-	return append(existing, add...)
+// UpdateAttestations returns a new state with the provided attestations.
+func (a *ActiveState) UpdateAttestations(attestations []*pb.AggregatedAttestation) *ActiveState {
```

Comment:
Should this be exported? I guess it has to because we are using it in blockchain service